### PR TITLE
chore: rotate tokens for e2e tests

### PR DESCRIPTION
**Pull request by Extheoisah**

### CODE REVIEW

File: e2e/utils/graphql.ts [hunk 1]
```diff
@@ -33,6 +33,27 @@ const createGaloyServerClient = (config) => (authToken) => {
   })
 }
 
+const authTokens = process.env.GALOY_TEST_TOKENS?.split(",") || []
+
+const randomizeTokens = (arr: string[]): string => {
```
Comment (nicolasburtey) [inline e2e/utils/graphql.ts:38]: this function is overly complicated.

here is chatGPT rewrite of the function:

```
const randomizeTokens = (arr: string[]): string => {
  const randomIndex = Math.floor(Math.random() * arr.length);
  return arr[randomIndex];
}
```

File: e2e/utils/graphql.ts [hunk 2]
```diff
@@ -33,10 +34,21 @@ const createGaloyServerClient = (config) => (authToken) => {
...
+  const randomIndex = Math.floor(Math.random() * arr.length)
+  return arr[randomIndex]
+}
+
+const authTokens = process.env.GALOY_TEST_TOKENS?.split(",") || []
+const firstAuthToken = randomizeTokens(authTokens)
+const secondAuthToken = process.env.GALOY_TOKEN
+
+console.log(firstAuthToken, secondAuthToken)
```
Comment (nicolasburtey) [inline e2e/utils/graphql.ts:46]: to be removed

File: e2e/utils/graphql.ts [hunk 3]
```diff
@@ -33,6 +33,27 @@ const createGaloyServerClient = (config) => (authToken) => {
...
+  for (let i = arr.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1))
+    ;[arr[i], arr[j]] = [arr[j], arr[i]]
+  }
+  return arr[0]
+}
+
+const getToken = () => {
+  const firstAuthToken = randomizeTokens(authTokens)
+  let secondAuthToken = randomizeTokens(authTokens)
```
[the remark this answers is not in the document]
Comment (Extheoisah) [reply e2e/utils/graphql.ts]: Yeah, resolved

File: e2e/utils/graphql.ts [hunk 4]
```diff
@@ -33,10 +34,16 @@ const createGaloyServerClient = (config) => (authToken) => {
...
-const authToken = process.env.GALOY_TOKEN
-const authTokenOther = process.env.GALOY_TOKEN_2
+const randomizeTokens = (arr: string[]): string => {
+  const randomIndex = Math.floor(Math.random() * arr.length)
+  return arr[randomIndex]
+}
+
+const authTokens = process.env.GALOY_TEST_TOKENS?.split(",") || []
+const secondAuthToken = process.env.GALOY_TOKEN_2 || ""
+export const firstAuthToken = randomizeTokens(authTokens)
```
Comment (UncleSamtoshi) [inline e2e/utils/graphql.ts:44]: I would rename these to be more specific,  such as `mobileUserToken` and secondAuthToken to be `recieverToken`

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

File: e2e/utils/graphql.ts [hunk 5]
```diff
@@ -33,10 +34,16 @@ const createGaloyServerClient = (config) => (authToken) => {
...
 
-const authToken = process.env.GALOY_TOKEN
-const authTokenOther = process.env.GALOY_TOKEN_2
+const randomizeTokens = (arr: string[]): string => {
+  const randomIndex = Math.floor(Math.random() * arr.length)
+  return arr[randomIndex]
+}
+
+const authTokens = process.env.GALOY_TEST_TOKENS?.split(",") || []
+const recieverToken = process.env.GALOY_TOKEN_2 || ""
```
Comment (UncleSamtoshi) [inline e2e/utils/graphql.ts:43]: Oh shoot my bad for the typo but it should be `receiver`

**Review by UncleSamtoshi — APPROVED** [review]

**Pull request merged**